File: playground/java/src/main/java/com/algolia/playground/Insights.java
package com.algolia.playground;

import com.algolia.api.InsightsClient;
import com.algolia.config.ClientOptions;
import com.algolia.config.LogLevel;
import com.algolia.exceptions.*;
import com.algolia.model.insights.*;
import io.github.cdimascio.dotenv.Dotenv;

public class Insights {

  public static void main(String[] args) throws Exception {
    var dotenv = Dotenv.configure().directory("../").load();
    var appId = dotenv.get("ALGOLIA_APPLICATION_ID");
    var apiKey = dotenv.get("ALGOLIA_SEARCH_KEY");
    var indexName = dotenv.get("SEARCH_INDEX");

    var options = new ClientOptions.Builder()
            .addAlgoliaAgentSegment("Playground")
            .setLogLevel(LogLevel.BASIC)
            .build();

    try(var client = new InsightsClient(appId, apiKey, options)) {
      var params = new InsightsEvents();
      var event = EventsItems.of(new ClickedObjectIDs()
              .setEventType(ClickEvent.CLICK)
              .setUserToken("user")
              .setIndex(indexName)
              .setEventName("test"));
      params.addEvents(event);
      var result = client.pushEvents(params);
      System.out.println(result);
    }
  }
}

File: playground/kotlin/src/main/kotlin/com/algolia/playground/Insights.kt
package com.algolia.playground

import com.algolia.client.api.InsightsClient
import com.algolia.client.configuration.ClientOptions
import com.algolia.client.model.insights.ClickEvent
import com.algolia.client.model.insights.ClickedObjectIDs
import com.algolia.client.model.insights.EventsItems
import com.algolia.client.model.insights.InsightsEvents
import io.github.cdimascio.dotenv.Dotenv
import io.ktor.client.plugins.logging.*
import kotlin.system.exitProcess

suspend fun main() {
    val dotenv = Dotenv.configure().directory("../").load()

    val client = InsightsClient(
        appId = dotenv["ALGOLIA_APPLICATION_ID"],
        apiKey = dotenv["ALGOLIA_SEARCH_KEY"],
        options = ClientOptions(logLevel = LogLevel.BODY),
    )
    val indexName = dotenv["SEARCH_INDEX"]

    val events = InsightsEvents(
        events = listOf(
            ClickedObjectIDs(
                eventType = ClickEvent.Click,
                eventName = "click",
                index = indexName,
                userToken = "playground_user",
                objectIDs = listOf("6445156")
            )
        )
    )

    val response = client.pushEvents(events)
    println(response)

    exitProcess(0)
}
